samples: add fuel gauge sample (#42)
NordicSemiconductor/npmx-zephyr: samples/fuel_gauge/README.rst

.. _npmx_fuel_gauge_sample:

> Fuel gauge

Reviewer: ```suggestion
Fuel Gauge
```

##########

.. contents::
   :local:
   :depth: 2

> The npmx fuel gauge sample demonstrates how to calculate the battery state of charge using the `nRF Fuel Gauge library`_ and `npmx drivers`_.

Reviewer: ```suggestion
The npmx Fuel Gauge sample demonstrates how to calculate the battery state of charge (SoC) with the use of the `nRF Fuel Gauge library`_ and `npmx drivers`_.
```

> The npmx Fuel Gauge sample demonstrates how to calculate the battery state of charge with the use of the `nRF Fuel Gauge library`_ and `npmx drivers`_.

Reviewer: ```suggestion
The npmx Fuel Gauge sample demonstrates how to calculate the battery State of Charge with the use of the `nRF Fuel Gauge library`_ and `npmx drivers`_.
```

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

The sample also requires an nPM1300 EK.

Overview
********

> This sample allows to calculate the state of charge, time to empty and time to full information from a battery connected to the nPM1300 PMIC.

Reviewer: ```suggestion
This sample allows to calculate the State of Charge, Time to Empty and Time to Full levels from a battery connected to the nPM1300 PMIC.
```

.. _CONFIG_TERMINATION_VOLTAGE_WARM:

CONFIG_TERMINATION_VOLTAGE_WARM
  This option changes the battery termination voltage in warm temperature.

.. _CONFIG_THERMISTOR_RESISTANCE:

CONFIG_THERMISTOR_RESISTANCE
  This option changes the thermistor nominal resistance.

.. _CONFIG_THERMISTOR_BETA:

CONFIG_THERMISTOR_BETA
  This option changes the thermistor beta value.

Building and running
********************

.. |sample path| replace:: :file:`samples/fuel_gauge`

.. include:: /includes/build_and_run.txt

Testing
=======

#. |connect_kit|
#. |connect_terminal|

If the initialization was successful, the terminal displays the following message with status information:

.. code-block:: console

   [00:00:00.000,000] <inf> fuel_gauge_main: PMIC device OK.
   [00:00:00.000,000] <inf> fuel_gauge_main: Fuel gauge OK.
   [00:00:00.000,000] <inf> fuel_gauge: V: 4.121, I: -0.127, T: 23.26, SoC: 87.63, TTE: nan, TTF: 2373

.. _table::
   :widths: auto

   ======  ===============  ==================================================
   Symbol  Description      Units
   ======  ===============  ==================================================
   V       Battery voltage  Volts

>    I       Current          Amps (negative for charge, positive for discharge)

Reviewer: ```suggestion
   I       Current          Amperes (negative for charge, positive for discharge)
```
